core/services/registrysyncer/syncer_test.go

```diff
@@ -321,6 +323,131 @@ func TestReader_Integration(t *testing.T) {
+
+	_, err := reg.AddCapabilities(owner, []kcr.CapabilitiesRegistryCapability{writeChainCapability})
+	require.NoError(t, err, "AddCapability failed for %s", writeChainCapability.LabelledName)
+	sim.Commit()
+
+	cid, err := reg.GetHashedCapabilityId(&bind.CallOpts{}, writeChainCapability.LabelledName, writeChainCapability.Version)
+	require.NoError(t, err)
+
+	_, err = reg.AddNodeOperators(owner, []kcr.CapabilitiesRegistryNodeOperator{
+		{
+			Admin: owner.From,
+			Name:  "TEST_NOP",
+		},
+	})
+	require.NoError(t, err)
+
+	nodeSet := [][32]byte{
+		randomWord(),
+		randomWord(),
+		randomWord(),
+	}
+
+	signersSet := [][32]byte{
+		randomWord(),
+		randomWord(),
+		randomWord(),
+	}
+
+	nodes := []kcr.CapabilitiesRegistryNodeParams{
+		{
+			// The first NodeOperatorId has id 1 since the id is auto-incrementing.
+			NodeOperatorId:      uint32(1),
+			Signer:              signersSet[0],
+			P2pId:               nodeSet[0],
+			HashedCapabilityIds: [][32]byte{cid},
+		},
+		{
+			// The first NodeOperatorId has id 1 since the id is auto-incrementing.
+			NodeOperatorId:      uint32(1),
+			Signer:              signersSet[1],
+			P2pId:               nodeSet[1],
+			HashedCapabilityIds: [][32]byte{cid},
+		},
+		{
+			// The first NodeOperatorId has id 1 since the id is auto-incrementing.
+			NodeOperatorId:      uint32(1),
+			Signer:              signersSet[2],
+			P2pId:               nodeSet[2],
+			HashedCapabilityIds: [][32]byte{cid},
+		},
+	}
+	_, err = reg.AddNodes(owner, nodes)
+	require.NoError(t, err)
+
+	config := &capabilitiespb.CapabilityConfig{
+		DefaultConfig: values.Proto(values.EmptyMap()).GetMapValue(),
+		RemoteConfig: &capabilitiespb.CapabilityConfig_RemoteTriggerConfig{
+			RemoteTriggerConfig: &capabilitiespb.RemoteTriggerConfig{
+				RegistrationRefresh: durationpb.New(20 * time.Second),
+				RegistrationExpiry:  durationpb.New(60 * time.Second),
+				// F + 1
+				MinResponsesToAggregate: uint32(1) + 1,
+			},
+		},
+	}
+	configb, err := proto.Marshal(config)
+	if err != nil {
+		require.NoError(t, err)
+	}
+
+	cfgs := []kcr.CapabilitiesRegistryCapabilityConfiguration{
+		{
+			CapabilityId: cid,
+			Config:       configb,
+		},
+	}
+	_, err = reg.AddDON(
+		owner,
+		nodeSet,
+		cfgs,
+		true,
+		true,
+		1,
+	)
+	sim.Commit()
+
+	require.NoError(t, err)
+
+	db := pgtest.NewSqlxDB(t)
+	wrapper := mocks.NewPeerWrapper(t)
+	factory := newContractReaderFactory(t, sim)
+	syncer, err := New(logger.TestLogger(t), wrapper, factory, regAddress.Hex(), db)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		syncer.Close()
+	})
+
+	l := &launcher{}
+	syncer.AddLauncher(l)
+
+	go func() {
+		// updateChan will be closed on cleanup
+		syncer.updateStateLoop()
+	}()
+
+	err = syncer.sync(ctx, false) // should store the data into the DB
+	require.NoError(t, err)
+	s := l.localRegistry
+	time.Sleep(1 * time.Second) // TODO: figure out another alternative to wait for the state to be stored in the DB
```

Comment:
@vyzaldysanchez This could become a flakey test; I would probably recommend just calling syncer.sync followed by the method inside the updater loop that saves it into the database.

Alternatively, you can introduce a test-only channel that the update loop can send a message to when it's done.

Reply:
@vyzaldysanchez This doesn't look like it's been addressed?